presto-main/src/main/java/io/prestosql/server/security/jwt/JwkDecoder.java

```diff
@@ -52,84 +56,104 @@ private JwkDecoder() {}
                 .collect(toImmutableMap(JwkPublicKey::getKeyId, Function.identity()));
     }
 
-    public static Optional<? extends JwkPublicKey> tryDecodeJwkKey(Map<String, String> properties)
+    public static Optional<? extends JwkPublicKey> tryDecodeJwkKey(Key key)
     {
-        String keyId = properties.get("kid");
-        if (Strings.isNullOrEmpty(keyId)) {
-            // key id is required to index the key
+        // key id is required to index the key
+        if (key.getKid().filter(Predicate.not(String::isEmpty)).isEmpty()) {
```

Comment:
`if(key.getKid().map(String::isEmpty).orElse(true))` maybe? I find calculating the boolean from Optional state after filter a bit too convoluted for me.

Actually, why not inlining it? 


  ```
      Optional<String> kid = key.getKid().filter(not(String::isEmpty)));
      switch (key.getKty()) {
            case "RSA":  
                 return kid.map(k -> tryDecodeRsaKey(k, key));
            case "EC":
                 return kid.map(k -> tryDecodeEcKey(k, key))
            default:
                // ignore non unknown keys
                return Optional.empty();
        }
```